src/main/java/com/faforever/client/leaderboard/LeaderboardsController.java

```diff
@@ -1,158 +1,104 @@
 import org.springframework.context.annotation.Scope;
 import org.springframework.stereotype.Component;
 
-import java.util.stream.Collectors;
-
-import static javafx.collections.FXCollections.observableList;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
 
 
 @Slf4j
 @Component
-@Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
 @RequiredArgsConstructor
+@Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
 public class LeaderboardsController extends AbstractViewController<Node> {
-
+  private final EventBus eventBus;
+  private final I18n i18n;
   private final LeaderboardService leaderboardService;
   private final NotificationService notificationService;
-  private final I18n i18n;
-  public Pane leaderboardRoot;
-  public TableColumn<LeaderboardEntryBean, Number> rankColumn;
-  public TableColumn<LeaderboardEntryBean, String> nameColumn;
-  public TableColumn<LeaderboardEntryBean, Number> winLossColumn;
-  public TableColumn<LeaderboardEntryBean, Number> gamesPlayedColumn;
-  public TableColumn<LeaderboardEntryBean, Number> ratingColumn;
-  public TableView<LeaderboardEntryBean> ratingTable;
-  public ComboBox<LeaderboardBean> leaderboardComboBox;
-  public TextField searchTextField;
-  public Pane connectionProgressPane;
-  public Pane contentPane;
+  private final UiService uiService;
+
+  public TabPane leaderboardRoot;
 
-  @VisibleForTesting
-  protected AutoCompletionBinding<String> usernamesAutoCompletion;
+  private boolean isHandlingEvent;
+  private final List<LeaderboardController> controllers = new ArrayList<>();
+  private Tab lastTab;
+
+  @Override
+  public Node getRoot() {
+    return leaderboardRoot;
+  }
 
   @Override
   public void initialize() {
-    super.initialize();
-    leaderboardService.getLeaderboards().thenApply(leaderboards -> {
-      JavaFxUtil.runLater(() -> {
-        leaderboardComboBox.getItems().clear();
-        leaderboardComboBox.setConverter(leaderboardStringConverter());
-        leaderboardComboBox.getItems().addAll(leaderboards);
-        leaderboardComboBox.getSelectionModel().selectFirst();
-      });
+    leaderboardService.getLeagues().thenAccept(leagues -> leagues.forEach(league ->
+        leaderboardService.getLatestSeason(league.getId()).thenAccept(season -> {
+          LeaderboardController controller = uiService.loadFxml("theme/leaderboard/leaderboard.fxml");
+          controller.setSeason(season);
+          controller.getRoot().setText(i18n.get(String.format("leaderboard.%s", league.getTechnicalName())));
+          controller.getRoot().setUserData(league.getId());
+          JavaFxUtil.runLater(() -> leaderboardRoot.getTabs().add(controller.getRoot()));
+          controllers.add(controller);
+        }).thenRun(() -> {
+          if (controllers.isEmpty()) {
+            log.info("Api returned no leagues");
+            notificationService.addImmediateErrorNotification(null, "leaderboard.noLeaderboards");
```

Comment:
Getting no leagues from the api means that something is seriously wrong, so I find a warning appropriate. Do you have ideas what we could display on the page if we can't load it normally?

Reply:
Yes I just think that the current message could imply that there just are no leagues rather than that something went wrong while loading the leagues.

As far as display maybe something like a simple uh-oh page message perhaps.

This would allow us to refresh the page again at least as well.

Reply:
The addImmediateWarnNotification displays an uh-oh message. So that is what you had in mind?

Reply:
Yeah that works as well